parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -175,12 +188,53 @@ pub mod pallet {
 	pub(super) type ExecutionHeaders<T: Config> =
 		StorageMap<_, Identity, H256, ExecutionHeader, OptionQuery>;
 
+	/// Execution headers ring buffer map implementation
+
+	/// Index storage for execution header ring buffer map
+	#[pallet::storage]
+	pub(crate) type ExecutionHeaderBufferIndex<T: Config> = StorageValue<_, u64, ValueQuery>;
+
+	/// Intermediate storage for execution header mapping
+	#[pallet::storage]
+	pub(crate) type ExecutionHeaderMapping<T: Config> =
+		StorageMap<_, Identity, u64, H256, ValueQuery>;
+
+	/// Ring buffer Map for Execution header
+	pub(crate) type ExecutionHeaderRingBufferMap<T> = RingBufferMapImpl<
```

Comment:
I don't think its worth the extra complexity to prune execution headers. For sync committees it makes sense, since they are huge.

I would rather we remove unnecessary fields from `ExecutionHeader`. All we really need is the `receipts_root` and `storage_root`.

Also see this discussion here for the Polkadot<->Kusama bridge, where they take a similar approach: https://github.com/Snowfork/cumulus/blob/c9cca565af5f165db5fbea9a149f9594c9a76b6e/bridges/modules/parachains/src/lib.rs#L187

Reply:
On second thoughts, since `StorageMap` and `StorageDoubleMap` support ring buffers, is ok to use it for execution headers.

Reply:
@vgeddes 
> I don't think its worth the extra complexity to prune execution headers. 
Agree but for another reason [here](https://linear.app/snowfork/issue/SNO-356#comment-4074dc26)

> since StorageMap and StorageDoubleMap support ring buffers
Really? AFAIK it does not really support. Some tests [here](https://github.com/Snowfork/snowbridge/commit/c9754f26765b23fb8841513857156824e8f3c0e9)

Reply:
See my comment here: https://github.com/Snowfork/snowbridge/pull/815#discussion_r1179964218

Reply:
@vgeddes  Regarding `ExecutionHeader` we do sync large number of execution headers compared to `SyncCommittees`. So, while object size is small overall the map of `ExecutionHeaders` occupies somewhat similar space as `SyncCommittees` map.

Reply:
Agree, I'm fine we keep execution headers pruning in this PR, [benchmark](https://github.com/Snowfork/cumulus/pull/20/files#diff-db08d74f3fe259c221df3ac86f73c1e06ebafc90e2907e95b98ff9cf71018ced) shows very limited overhead can be safely ignored.  For the long run I look forward [eip-4788](https://eips.ethereum.org/EIPS/eip-4788) then perhaps we can remove importing execution headers completely.

Reply:
Ok lets prune execution headers then 👍